mapper/orm/src/main/java/org/hibernate/search/mapper/orm/session/SearchSession.java

```diff
@@ -94,4 +94,15 @@
 	 */
 	Session toOrmSession();
 
+	/**
+	 * Set the {@link AutomaticIndexingSynchronizationStrategy} to use for this session.
+	 * <p>
+	 * Behavior is undefined if called while entity changes are pending:
+	 * be sure to call this only just after creating a session,
+	 * or just after committing a transaction.
+	 *
+	 * @param synchronizationStrategy The synchronization strategy to use
+	 * @see AutomaticIndexingSynchronizationStrategy
+	 */
+	void setAutomaticIndexingSynchronizationStrategy(AutomaticIndexingSynchronizationStrategy synchronizationStrategy);
```

Comment:
I'm asking to myself. Maybe it would be better to make this value immutable for the session instance? Since I don't get the value on reuse the same session with different overridden strategies.

Reply:
Ideally, yes. In practice, it's not easy to do so.

In order to make this value immutable for the session instance, we would have to set it upon session creation. There are several problems with that:

1. Most users do not create the session themselves, and get it injected instead. They can still customize its creation, but it's not always obvious.
2. Even for us, technically, it's complicated. In order to add our own settings upon session creation, we would have to either create a wrapper for the session builder, which is a bit dodgy and is what we've been trying to avoid when we moved away from the ORM session interface as a superclass of our own. Or we could try to use extension points in `SessionBuilder`... but there are none, not even a `hint(String, Object)` method. Or we could allow to change the SearchSession associated to a given ORM Session, but that would boil down to mutating the session.